scipy/special/basic.py

```diff
@@ -85,13 +82,6 @@ def __dir__():
 
 
 def __getattr__(name):
-    if name not in __all__:
-        raise AttributeError(
-            "scipy.special.basic is deprecated and has no attribute "
-            f"{name}. Try looking in scipy.special instead.")
-
-    warnings.warn(f"Please use `{name}` from the `scipy.special` namespace, "
-                  "the `scipy.special.basic` namespace is deprecated.",
-                  category=DeprecationWarning, stacklevel=2)
-
-    return getattr(_basic, name)
+    return _sub_module_deprecation(sub_package="special", module="basic",
+                            private_modules=["_basic", "_ufuncs"], all=__all__,
+                            attribute=name)
```

Comment:
Just a style nit but I think you're fine to indent like this and stay within the 88 char limit. As far as I have looked all of the other uses of `_sub_module_deprecation` have indented like this.
```suggestion
    return _sub_module_deprecation(sub_package="special", module="basic",
                                   private_modules=["_basic", "_ufuncs"], all=__all__,
                                   attribute=name)
```